mobile/memrpc.go

```diff
@@ -98,13 +105,95 @@ func getClient() (lnrpc.LightningClient, context.Context, func(), error) {
 	if err != nil {
 		return nil, nil, nil, err
 	}
-	client := lnrpc.NewLightningClient(clientConn)
+	lightningClient := lnrpc.NewLightningClient(clientConn)
+
 	ctx, cancel := context.WithCancel(context.Background())
-	return client, ctx, cancel, nil
+	return lightningClient, ctx, cancel, nil
+}
+
+// getUnlockerClient returns a client connection to the server listening on unlockerLis.
+func getUnlockerClient() (lnrpc.WalletUnlockerClient, context.Context, func(), error) {
+	conn, err := unlockerLis.Dial()
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	clientConn, err := grpc.Dial("",
+		grpc.WithDialer(func(target string,
+			timeout time.Duration) (net.Conn, error) {
+			return conn, nil
+		}),
+		grpc.WithInsecure(),
+		grpc.WithBackoffMaxDelay(10*time.Second),
+	)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	unlockerClient := lnrpc.NewWalletUnlockerClient(clientConn)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	return unlockerClient, ctx, cancel, nil
+}
+
+// unlockOnceHandler is a struct used to call the daemon's RPC interface on
+// WalletUnlocker methods where only one request and one response is expected.
+type unlockOnceHandler struct {
+	// newProto returns an empty struct for the desired grpc request.
+	newProto func() proto.Message
+
+	// getSync calls the desired method on the given client in a
+	// blocking matter.
+	getSync func(context.Context, lnrpc.WalletUnlockerClient,
+		proto.Message) (proto.Message, error)
+}
+
+// start executes the RPC call specified by this onceHandler using the
+// specified serialized msg request.
+func (s *unlockOnceHandler) start(msg []byte, callback Callback) {
```

Comment:
same, should try to reuse as much as possible :)

Reply:
Hey Thanks for looking at this! My struggle is with mixing the 2 return interface types together LightningClient & WalletUnlockerClient. Any go construct I should look at for this? Still a newbie at go for sure. I'll regardless revisit this in the next few days to see if I can come up with anything better to reduce redundancies. But any tips would be appreciated!